dlib/dnn/loss.h

```diff
@@ -367,6 +367,137 @@ namespace dlib
     template <typename SUBNET>
     using loss_multiclass_log = add_loss_layer<loss_multiclass_log_, SUBNET>;
 
+// ----------------------------------------------------------------------------------------
+
+    class loss_multiclass_log_weighted_ 
+    {
+    public:
+
+        struct weighted_label
```

Comment:
In that other class `label` is `uint16_t` (here `unsigned long`), but otherwise this is identical, right?

Should there instead be a common class which both can use? (You could make `label` be a template type.)

Reply:
Yes, it is identical. I was also thinking that maybe we could use a common class for both. I can do that :)

Reply:
Ok, I've done what you suggested. I was wondering if `weighted_label` is an appropriate name now that it's in the `dlib` namespace. Also if I should also add serialization support for it...